packages/playground/remote/src/lib/playground-mu-plugin/0-playground.php

```diff
@@ -91,6 +91,32 @@ function networking_disabled() {
 	<?php
 });
 
+/**
+ * Adds target="_blank" to external links when clicked to open them in a new tab.
+ * This prevents users from loading non-Playground pages inside the Playground iframe.
+ */
+function playground_add_target_blank_to_external_links() {
+	// Only run on frontend and admin pages, not during AJAX requests or CLI
+	if (empty($_SERVER['REQUEST_URI']) || wp_doing_ajax() || wp_doing_cron()) {
+		return;
+	}
+	?>
+	<script>
+		document.addEventListener('click', e => {
+			const a = e.target.closest('a[href]');
+			if (!a) return;                                 // not a link
+			if (new URL(a.href, location).origin !== location.origin) {
+				a.target = '_blank';                        // open in new tab/window
+				a.rel ||= 'noopener';                       // cuts the opener reference
+			}
+		});
```

Comment:
I think this covers the 99% case, but I just want to mention that it won't work for keyboard navigation, for instance. It would probably be more robust to add the attribute to all links ahead of time:

```js
document.addEventListener('DOMContentLoaded', () => {
	document.querySelectorAll('a[href]').forEach(a => {
		const url = new URL(a.href, location);
		if (url.origin !== location.origin) {
			a.target = '_blank';
			a.rel ||= 'noopener';
		}
	});
});
```

Reply:
Oh, I see that's an experimental interface so probably not suitable.

Reply:
Good spot! It still wouldn't cover for dynamically added links, and I'm hesitant of adding mutation observers, but it still moves us further and covers more links – I like that.

Reply:
I also added a `focus` listener to cover keyboard navigation of dynamically added links